### ISSUE TITLE
regression: `set_hl` does not apply highlight

### ISSUE THREAD

**Issue by Alexdelia** [noib3/nvim-oxi#203]
## issue

`nvim_oxi::api::set_hl` doesn't apply expected color highlight

---

## test

```rs
nvim_oxi::api::set_hl(
    0,
    "Type",
    &SetHighlightOpts::builder()
        .bold(true)
        .foreground("#579dd6")
        .build(),
)?;
```

`:highlight Type` output:
```qml
Type           xxx cterm=bold gui=bold guifg=#579dd6
```
the output is correct

---

## problem

expected color:
![image](https://github.com/user-attachments/assets/44ad6477-347f-4dfc-8b13-06ff9ea1de38)

got color:
![image](https://github.com/user-attachments/assets/c17c6d94-aa01-4096-be46-ba845767068a)

---

## manual solution

after doing:
```qml
:highlight Type gui=bold
```
the color becomes the correct one

and the ouput of `:highlight Type` stays the same

---

## context

`nvim_oxi` version: a30581cf0019dcac7a982cf09afd3cd67777519b (older version are also affected)
- with features flag `["neovim-0-10"]`

`neovim` version:
```
NVIM v0.10.2
Build type: Release
LuaJIT 2.1.1713773202
```

my repo: https://github.com/Alexdelia/vity.nvim/tree/506908989c08e1ace1335b81a02c438dbe04bef6
code from test: https://github.com/Alexdelia/vity.nvim/blob/506908989c08e1ace1335b81a02c438dbe04bef6/src/highlight/code/type.rs#L6

---

## when it started

the issue started after doing the transition from `neovim` `0.9` to `0.10`

---

## extra

with the code:
```rs
nvim_oxi::api::set_hl(
    0,
    "Type",
    &SetHighlightOpts::builder()
        // .bold(true)
        .foreground("#579dd6")
        .build(),
)?;
```

the output of `:highlight Type` is:
```qml
Type           xxx cterm= gui= guifg=#579dd6
```

while I would expect
```qml
Type           xxx guifg=#579dd6
```

**Comment by noib3** [comment]
Can you replicate it by building against `main`? This is likely already resolved by https://github.com/noib3/nvim-oxi/issues/187.

**Comment by Alexdelia** [comment]
> Can you replicate it by building against `main`? This is likely already resolved by #187.

a30581cf0019dcac7a982cf09afd3cd67777519b is the latest `commit` on `main`, no?

**Comment by Alexdelia** [comment]
I can test on any `commit` you want me to if you would like me to try right after the `PR` got `merged` or something else

**Comment by Alexdelia** [comment]
@noib3, just tried on b986c4951db0aa21c6627bf25c5d9e3e931ade07 (which is the `merge` `commit` that solve #187) and it works

which means that it's a regression

**Comment by Alexdelia** [comment]
I will do a few more test to make sure that I did not mess up when I tried the first time

**Comment by noib3** [comment]
> nvim_oxi version: https://github.com/noib3/nvim-oxi/commit/a30581cf0019dcac7a982cf09afd3cd67777519b (older version are also affected)

Oh sorry, I didn't see this.

But I'm not sure I understood what the problem is. Can you write a test that fails in Neovim 0.10?

**Comment by noib3** [comment]
Can you try updating `nvim-oxi` to the latest commit? This might've been fixed by 3c87723.

**Comment by Alexdelia** [comment]
I'm back

I tested again a30581cf0019dcac7a982cf09afd3cd67777519b to be sure I didn't mess any previous test and the problem was indeed present

I then tested on the latest `commit` as of right now 61cc49037004c7cc21a0e44d69388d31315bc26b and you are correct, it has been fixed

thanks a lot you are incredible
